### ISSUE TITLE
(🐞) false error with `type[X]` in stub file when `python-version` <= 3.8

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#11926]
**To Reproduce**

`__init__.pyi`:
```py
l: list[int]  # no error
i: type[int]  # error
```

**Expected Behavior**
Because this is just a stub and not being executed, `type[X]` should be valid and not raise any errors.

**Actual Behavior**
`error: "type" expects no type arguments, but 1 given`

**Your Environment**
- Mypy version used: 0.930
- Mypy command-line flags: `--python-version 3.9`
- Python version used: 3.11

**Comment by hauntsaninja** [comment]
This is fixed on master
